### DISCUSSION TITLE
Where are the published packages stored on my local machine?

### DISCUSSION THREAD

**Discussion by tearingItUp786** [verdaccio/verdaccio#2024] [Q&A] [answered]
Hey all, 

I'm running through a workshop on frontendmasters and we're using verdaccio to set up our own local npm proxy server. I did notice one thing though, if I uninstall and reinstall verdaccio, the packages I published are still around... I can't find where these published packages are saved on my local machine. Does anyone have any ideas?

I'm using a mac os and I'm installing packages globally with volta.

**Comment by DanielRuf** [comment, marked as answer]
See https://verdaccio.org/docs/en/cli#default-storage-location

> We use the $XDG_DATA_HOME environment by variable default to locate the storage by default which should be the same as $HOME/.local/share. If you are using a custom storage, this location is irrelevant.

So by default it is a folder `storage` in the path of the configuration which is shown when verdaccio starts.

**Reply by juanpicado** [reply]
In addition, for Windows is on `AppData` folder https://www.howtogeek.com/318177/what-is-the-appdata-folder-in-windows/

**Reply by tearingItUp786** [reply]
I tried to echo out $XDG_DATA_HOME in my terminal and I got an empty output so it's hard for me to figure out where this is. 

For context, I installed verdaccio globally with [volta](https://volta.sh/)

**Reply by DanielRuf** [reply]
When XDG_DATA_HOME is not set, `~/.local/share` is used.

See https://unix.stackexchange.com/a/39062/159964

> $XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored. If $XDG_DATA_HOME is either not set or empty, a default equal to $HOME/.local/share should be used.

So the path for data (not config) is `~/.local/share`.
For me the storage folder is at `/Users/danielruf/.local/share/verdaccio/storage`.
